tests/test_api.py

```diff
@@ -563,3 +554,39 @@ def test_history():
     assert isinstance(estimator.history_, dict)
     assert all(isinstance(k, str) for k in estimator.history_.keys())
     assert all(isinstance(v, list) for v in estimator.history_.values())
+
+
+def test_compile_model_from_params():
+    """Tests that if build_fn returns an un-compiled model,
+    the __init__ parameters will be used to compile it
+    and that if build_fn returns a compiled model
+    it is not re-compiled.
+    """
+    # Load data
+    data = load_boston()
+    X, y = data.data[:100], data.target[:100]
+
+    losses = ("mean_squared_error", "mean_absolute_error")
+
+    # build_fn that does not compile
+    def build_fn(myloss=None):
+        model = Sequential()
+        model.add(keras.layers.Dense(X.shape[1], input_shape=(X.shape[1],)))
+        model.add(keras.layers.Activation("relu"))
+        model.add(keras.layers.Dense(1))
+        model.add(keras.layers.Activation("linear"))
+        if myloss:
+            model.compile(loss=myloss)
+        return model
+
+    for loss in losses:
+        estimator = KerasRegressor(build_fn=build_fn, loss=loss, myloss=None)
```

Comment:
Could you add an explanation that `myloss=None` returns an uncompiled model? Or maybe a parameter `model__compile=False` and then adapt `build_fn` accordingly.

Reply:
I'll rename to `compile_with_loss=None` and add a comment.

Reply:
> I'd suggest raising an error in `BaseWrapper.fit` if the loss/optimizer hasn't been configured. TF throws an error on "ValueError: No gradients provided for any variable", which isn't super clear. I think that belongs in this PR (not #79).

By "hasn't been configured", do you think a `self.model_.loss is not None` check will suffice?

> Is #73 up to date?

It is largely up to date, but I just looked at the rendered version and it does need some work on typos and such.